### PR TITLE
TT Input embeddings

### PR DESCRIPTION
**Pull request by ntarafdar**
Single core implementation for TT Embeddings.
Some small fixes were required to enable UInt32 on device support required for the input indices.
@arakhmati could you look at the small changes I made to the tensor_impl files, it is just adding to a few if condtions.
@davorchap could you have a look at the small change I made to the tennis header, this is to add UIN32 support, needed to expand the enumeration.

Also @arakhmati could you have a look at the test_embeddings.py and the embeddings kernel to see if there are any other issues with how I used UINT32, its rather new, but I did test this so this should be okay.

And everyone else if possible please look at the op it self in embeddings_op and embeddings.cpp

Edit: This is good to be looked at. Thank you all in advance

Edit 2: This is back to draft, found a bug for larger tables.

Edit 3: Issues fixed.

### CODE REVIEW

**Review by arakhmati — APPROVED** [review]
LGTM!

**Review by tt-aho — CHANGES_REQUESTED** [review]

**Review by tt-aho — CHANGES_REQUESTED** [review]

**Review by tt-aho — APPROVED** [review]
lgtm once pipelines pass. Just squash some of the commits together before merging.

**Pull request merged**
